Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+            currentExtentEndTimeLabel.font = currentExtentLabelFont as CFTypeRef
+            currentExtentEndTimeLabel.fontSize = currentExtentLabelFont.pointSize
+        }
+    }
+    
+    /*
+     The date formatter used for full extent dates.
+     */
+    public var currentExtentDateFormatter: DateFormatter = {
+        let formatter: DateFormatter = DateFormatter()
+        formatter.locale = Locale.current
+        formatter.timeZone = TimeZone.current
+        formatter.dateStyle = .medium
+        return formatter
+    }()
+    
+    /*
+     A Boolean value that indicates whether the start time of the currentExtent can be
+     manipulated through user interaction or moves when time is being animated.
+     */
+    public var isStartTimePinned = false {
+        didSet {
+            lowerThumbLayer.isPinned = isStartTimePinned
+        }
+    }
+    
+    /*
+     A Boolean value that indicates whether the end time of the currentExtent can be
+     manipulated through user interaction or moves when time is being animated.
+     */
+    public var isEndTimePinned = false {
+        didSet {
+            if isRangeEnabled {
+                upperThumbLayer.isPinned = isEndTimePinned
+            }
+            else {
+                isStartTimePinned = isEndTimePinned
+                lowerThumbLayer.isPinned = isEndTimePinned
+                upperThumbLayer.isPinned = isEndTimePinned
+            }
+        }
+    }
+    
+    // MARK: Full Extent Properties
+    
+    /*
+     The full extent time window on the slider.
+     */
+    public var fullExtent: AGSTimeExtent? {
+        didSet {
+            //
+            // Set current extent if it's nil.
+            if currentExtent == nil, let fullExtent = fullExtent {
+                currentExtent = fullExtent
+            }
+            else if fullExtent == nil {
+                timeSteps?.removeAll()
+                tickMarks?.removeAll()
+                removeTickMarkLabels()
+                currentExtent = fullExtent
+            }
+            else {
+                //
+                // It is possible that the current extent times are outside of the range of
+                // new full extent times. Adjust and sanp them to the tick marks.
+                if let startTime = currentExtentStartTime, let endTime = currentExtentEndTime {
+                    updateCurrentExtentStartTime(startTime)
+                    updateCurrentExtentEndTime(endTime)
+                }
+            }
+            refresh()
+        }
+    }
+    
+    /*
+     The color used to tint the full extent border of the slider.
+     */
+    public var fullExtentBorderColor: UIColor = UIColor.black {
+        didSet {
+            trackLayer.setNeedsDisplay()
+        }
+    }
+    
+    /*
+     The border width of the full extent of the slider.
+     */
+    public var fullExtentBorderWidth: Double = 1.0 {
```

Comment:
It might make more sense for this to be a `CGFloat`. This seems to be a drawing-related property and making it a CGFloat would prevent having to cast it later while drawing.

Reply:
fixed